nilearn/plotting/displays.py

```diff
@@ -237,7 +245,27 @@ def transform_to_2d(self, data, affine):
                 The affine of the volume
 
         """
-        max_axis = 'xyz'.index(self.direction)
+        if self.direction in 'xlr':
+            max_axis = 0
+        else:
+            max_axis = '.yz'.index(self.direction)
+
+        # set unselected brain hemisphere activations to 0
+
+        if self.direction == 'l':
+            x_center, _, _, _ = np.dot(np.linalg.inv(affine),
+                                       np.array([0, 0, 0, 1]))
+            data_selection = data[int(x_center):, :, :]
+        elif self.direction == 'r':
+            x_center, _, _, _ = np.dot(np.linalg.inv(affine),
+                                       np.array([0, 0, 0, 1]))
+            data_selection = data[:int(x_center), :, :]
+        else:
+            data_selection = data
+
+        # We need to make sure data_selection is not empty in the x axis
+        if data_selection.shape[0] == 0:
+            data_selection = data
```

Comment:
I am not sure this is a good way to avoid a crash during smoke testing. Basically this will avoid the function crashing if the img passed has an affine that maps coordinate 0mm to a voxel index out of the array. How should handle that case. Actually raise a warning/error to the user?